pkg/server/sessions/session.go

```diff
@@ -53,12 +55,18 @@ type Session struct {
 	creationTime             time.Time
 	lastActivityTime         time.Time
 	transactions             map[string]transactions.Transaction
-	paginatedDocumentReaders map[string]*PaginatedDocumentReader // map from searchID to sql.RowReader objects
+	paginatedDocumentReaders *cache.LRUCache // track searchID to sql.RowReader objects
 	log                      logger.Logger
 }
 
-func NewSession(sessionID string, user *auth.User, db database.DB, log logger.Logger) *Session {
+func NewSession(sessionID string, user *auth.User, db database.DB, maxActiveSnapshots int, log logger.Logger) *Session {
+	if maxActiveSnapshots == 0 {
+		maxActiveSnapshots = store.DefaultMaxActiveTransactions
+	}
+
 	now := time.Now()
+	lruCache, _ := cache.NewLRUCache(maxActiveSnapshots)
```

Comment:
I see, I was assuming wrongly then. The problem is that it would mean there could only be maxActiveSnapshots number of read queries which can be open. Curtails it a bit, but no other option then

Reply:
cache being held by manager implies it would have to be maintained per database